R/paged.R

```diff
@@ -163,10 +163,107 @@ html_format = function(
       pagedown_dependency(xfun::with_ext(css2, '.css'), .pagedjs, .test)
     ))
   }
-  html_document2(
+  format = html_document2(
     ..., self_contained = self_contained, anchor_sections = anchor_sections, mathjax = mathjax, css = css,
     template = template, pandoc_args = pandoc_args
   )
+  if (isTRUE(.pagedjs)) format$knitr$opts_chunk[['render']] = paged_render(self_contained)
+  iframe_file(reset = TRUE)
+  format
+}
+
+paged_render = function(self_contained) {
+  function(x, options, ...) {
+    if (inherits(x, 'htmlwidget')) {
+      class(x) = c('iframehtmlwidget', class(x))
+    }
+    knitr::knit_print(x, options, ..., self_contained = self_contained)
+  }
+}
+
+knit_print.iframehtmlwidget = function(x, options, ..., self_contained) {
+  class(x) = tail(class(x), -1)
+  d = options$fig.path
+  if (!dir.exists(d)) {
+    dir.create(d, recursive = TRUE)
+    if (self_contained) on.exit({
+      unlink(d, recursive = TRUE) # doesn't work, don't understand why
+    }, add = TRUE)
```

Comment:
I'm not sure why it doesn't work, but does it help if you convert it to an absolute path? i.e.

```suggestion
    d = normalizePath(d)
    if (self_contained) on.exit(unlink(d, recursive = TRUE), add = TRUE)
```